### PR TITLE
test/dss: update tests to make them usable again, and run them

### PR DESCRIPTION
**Pull request by ggouaillardet**

### CODE REVIEW

File: test/dss/dss_copy.c [hunk 1]
```diff
@@ -57,97 +59,97 @@ int main (int argc, char* argv[])
...
     if (test13()) {
         fprintf(test_out, "Test13 succeeded\n");
-    }
-    else {
+    } else {
         fprintf(test_out, "opal_dss test13 failed\n");
+        ret = 13;
     }
 
     fclose(test_out);
```
Comment (jsquyres) [inline test/dss/dss_copy.c]: Shouldn't the return after this be updated to `return ret`?

**Comment by jsquyres** [comment]
@ggouaillardet I get the following compiler warnings when compiling this:

```
dss_cmp.c: In function ‘test12’:
dss_cmp.c:566:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (i=0; i<v1.size; i++) v1.bytes[i] = i;
                ^
dss_cmp.c:570:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (i=0; i<v2.size; i++) v2.bytes[i] = v1.bytes[i];
                ^
dss_print.c: In function ‘test12’:
dss_print.c:466:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (i=0; i<v2.size; i++) v2.bytes[i] = i;
                ^
```

Then the `dss_cmp` test fails for me:

```sh
[4:21] savbu-usnic-a:~/g/o/t/dss (ggouaillardet-topic/dss_tests)
$ ./dss_cmp
executing test1
Test1 succeeded
executing test2
Test2 succeeded
executing test3
Test3 succeeded
executing test4
Test4 succeeded
executing test5
Test5 succeeded
executing test6
Test6 succeeded
executing test7
Test7 succeeded
executing test8
Test8 succeeded
executing test9
Test9 succeeded
executing test11
Test11 succeeded
executing test12
Test12 succeeded
executing test13
opal_dss.compare failed for value1 greater
opal_dss test13 failed
```

**Comment by ggouaillardet** [comment]
I will add the missing return statement

I solved the dss_cmp bug and git push --force

I will double check this tomorrow, are you using the latest commits ?

**Comment by jsquyres** [comment]
I used [hub](https://github.com/github/hub) to check out this PR, so it should have gotten the latest of whatever is on this PR.

**Comment by ggouaillardet** [comment]
@jsquyres i updated the PR, but did not see anything wrong with dss_cmp

can you please give it an other try ?

**Comment by jsquyres** [comment]
Passes for me now.  :+1:

**Pull request merged**